rooms.js

```diff
@@ -1451,11 +1451,13 @@ Rooms.createChatRoom = function (roomid, title, data) {
 Rooms.createBattle = function (format, options) {
 	const p1 = options.p1;
 	const p2 = options.p2;
-	if (p1 === p2) throw new Error(`Players can't battle themselves`);
-	if (!p1) throw new Error(`p1 required`);
-	if (!p2) throw new Error(`p2 required`);
-	Ladders.matchmaker.cancelSearch(p1);
-	Ladders.matchmaker.cancelSearch(p2);
+	if (p1) Ladders.matchmaker.cancelSearch(p1);
+	if (p2) Ladders.matchmaker.cancelSearch(p2);
+	if (p1 === p2) {
+		p1.popup("You can't battle your own account. Please use something like Private Browsing to battle yourself.");
```

Comment:
This is probably the wrong place for this, even. This should probably go in the challenge code.